sdk/iot/hub/samples/src/paho_iot_hub_twin_example.c

```diff
@@ -160,51 +166,136 @@ static az_result read_configuration_and_init_client()
+  while (!az_span_is_content_equal(token_member.name, version_name))
+  {
+    if (az_span_is_content_equal(token_member.name, reported_property_name))
+    {
+      reported_property_value = (int32_t)token_member.token._internal.number;
+      AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(&jb, reported_property_name));
+      AZ_RETURN_IF_FAILED(az_json_builder_append_int32_number(&jb, reported_property_value));
+
+      AZ_RETURN_IF_FAILED(az_json_builder_append_end_object(&jb));
+
+      reported_payload = az_json_builder_get_json(&jb);
+
+      printf("Updating device_count reported property to service.\n");
+      printf(
+          "Payload: %.*s\n", az_span_size(reported_payload), (char*)az_span_ptr(reported_payload));
+
+      // Publish the reported property payload to IoT Hub
+      int rc;
+      if ((rc = MQTTClient_publish(
+               mqtt_client,
+               reported_property_topic,
+               az_span_size(reported_payload),
+               az_span_ptr(reported_payload),
+               0,
+               0,
+               NULL))
+          != MQTTCLIENT_SUCCESS)
+      {
+        printf("Failed to publish reported property JSON. Return code %d.\n", rc);
+        // Need an MQTT facility error, not sure what to create for error name.
+        // return
+      }
+      return AZ_OK;
+    }
+    AZ_RETURN_IF_FAILED(az_json_parser_parse_token_member(&jp, &token_member));
+  }
+
+  printf("Did not find \"device_count\" in desired property payload.\n");
+  return AZ_OK;
+}
+
 static int on_received(void* context, char* topicName, int topicLen, MQTTClient_message* message)
 {
   (void)context;
 
+  printf("Received a message from service.\n");
+  printf("Topic: %s\n", topicName);
+
   if (topicLen == 0)
   {
     // The length of the topic if there are one or more NULL characters embedded in topicName,
     // otherwise topicLen is 0.
     topicLen = (int)strlen(topicName);
   }
-
-  printf("Topic: %s\n", topicName);
-
   az_span topic_span = az_span_init((uint8_t*)topicName, topicLen);
 
   // Parse the incoming message topic and check to make sure it is a twin message
   az_iot_hub_client_twin_response twin_response;
-  if (az_succeeded(
-          az_iot_hub_client_twin_parse_received_topic(&client, topic_span, &twin_response)))
+  if (az_failed(az_iot_hub_client_twin_parse_received_topic(&client, topic_span, &twin_response)))
   {
-    printf("Twin Message Arrived\n");
+    printf("Topic is not a twin message.\n");
+  }
+  else
+  {
+    printf("Topic is a twin message.\n");
 
-    // Determine what type of incoming twin message this is. Print relevant data for the message.
+    // Determine type of incoming twin message. Print relevant data for the message.
     switch (twin_response.response_type)
     {
-      // A response from a twin GET publish message with the twin document as a payload.
+      // Type: A service response to a twin GET publish message.
+      // Payload: The JSON twin document.
```

Comment:
Trying to respond to your question above about device_count updating.
Is the updating not happening on the service?
Also, when 'r' is chosen, the device_count current value will be reported to the service and THEN it will increment.  So an additional 'r' is needed to send the incremented device_count to the service as a reported property.
This threw me off at first, as I was expecting to see the reported property be the incremented value.  Do we want to change this to be such?

Reply:
Scenario I tried was booting up, not touching anything, and sending the `device_count` with some number. Nothing gets pinged back to the service on my run. Could you try it on yours to verify?

Reply:
This should be fixed now.  Thanks for the catch.